ros/angel_system_nodes/angel_system_nodes/intent_detector.py

```diff
@@ -0,0 +1,140 @@
+    to a model-based approach in the neat-future.
+    '''
+
+    def __init__(self):
+        super().__init__(self.__class__.__name__)
+        self.log = self.get_logger()
+
+        parameter_names = [
+            UTTERANCES_TOPIC,
+            PARAM_EXPECT_USER_INTENT_TOPIC,
+            PARAM_INTERP_USER_INTENT_TOPIC,
+        ]
+        set_parameters = self.declare_parameters(
+            namespace="",
+            parameters=[(p,) for p in parameter_names],
+        )
+        # Check for not-set parameters
+        some_not_set = False
+        for p in set_parameters:
+            if p.type_ is rclpy.parameter.Parameter.Type.NOT_SET:
+                some_not_set = True
+                self.log.error(f"Parameter not set: {p.name}")
+        if some_not_set:
+            raise ValueError("Some parameters are not set.")    
+
+        self._utterances_topic = \
+            self.get_parameter(UTTERANCES_TOPIC).value
+        self._expect_uintent_topic = \
+            self.get_parameter(PARAM_EXPECT_USER_INTENT_TOPIC).value
+        self._interp_uintent_topic = \
+            self.get_parameter(PARAM_INTERP_USER_INTENT_TOPIC).value
+        self.log.info(f"Utterances topic: "
+                      f"({type(self._utterances_topic).__name__}) "
+                      f"{self._utterances_topic}")
+        self.log.info(f"Expected User Intent topic: "
+                      f"({type(self._expect_uintent_topic).__name__}) "
+                      f"{self._expect_uintent_topic}")
+        self.log.info(f"Interpreted User Intent topic: "
+                      f"({type(self._interp_uintent_topic).__name__}) "
+                      f"{self._interp_uintent_topic}")
+        
+        # TODO(derekahmed): Add internal queueing to reduce subscriber queue
+        # size to 1.
+        self.subscription = self.create_subscription(
+            Utterance,
+            self._utterances_topic,
+            self.listener_callback,
+            10)
+        
+        self._expected_publisher = self.create_publisher(
+            InterpretedAudioUserIntent,
+            self._expect_uintent_topic,
+            1
+        )
+
+        self._interp_publisher = self.create_publisher(
+            InterpretedAudioUserIntent,
+            self._interp_uintent_topic,
+            1
+        )
+
+    def listener_callback(self, msg):
+        log = self.get_logger()
+        intent_msg = InterpretedAudioUserIntent()
+        intent_msg.utterance_text = msg.value
+
+        lower_utterance = msg.value.lower()            
+        if self.contains_phrase(lower_utterance, NEXT_STEP_KEYPHRASES):
+            intent_msg.user_intent = LABELS[0]
+            intent_msg.confidence = 0.5
+        elif self.contains_phrase(lower_utterance, PREV_STEP_KEYPHRASES):
+            intent_msg.user_intent = LABELS[1]
+            intent_msg.confidence = 0.5
+        else:
+            intent_msg.user_intent = LABELS[2]
+            intent_msg.confidence = 0.5
```

Comment:
If the utterance does not contain any of the keyphrases, I think we could ignore it and return immediately.

I added the return statement locally, and it helped the ARUI not get bogged down with "Other" messages to confirm.

Reply:
Done.